### PR TITLE
Add targets limit per user

### PR DESCRIPTION
**Pull request by andresg4**
## TASKS
- [X] Added limit of targets per user.

### CODE REVIEW

File: app/models/target.rb [hunk 1]
```diff
@@ -10,4 +10,19 @@ class Target < ApplicationRecord
   validates :title, uniqueness: true
 
   validates :radius, numericality: { greater_than_or_equal_to: 0 }
+
+  validate :validate_on_create, on: :create
+
+  private
+
+  LIMIT_TARGETS = 10
```
Comment (matiasmansilla1989) [inline app/models/target.rb]: This constant should be almost on top of the file https://github.com/rootstrap/tech-guides/blob/master/ruby/rails.md#macro-style-methods

File: app/models/target.rb [hunk 2]
```diff
@@ -10,4 +10,19 @@ class Target < ApplicationRecord
...
+
+  def validate_on_create
+    return unless user
+
+    return unless user.targets.count >= LIMIT_TARGETS
```
Comment (matiasmansilla1989) [inline app/models/target.rb]: You can also do `return unless user || user.targets.count >= LIMIT_TARGETS`

File: app/models/target.rb [hunk 3]
```diff
@@ -10,4 +10,19 @@ class Target < ApplicationRecord
...
+
+    errors.clear
```
Comment (matiasmansilla1989) [inline app/models/target.rb]: I would remove this line and return all errors. The order in where validations are executed shouldn't affect which errors are returned.

File: app/models/target.rb [hunk 4]
```diff
@@ -10,4 +10,19 @@ class Target < ApplicationRecord
   validates :title, uniqueness: true
 
   validates :radius, numericality: { greater_than_or_equal_to: 0 }
+
+  validate :validate_on_create, on: :create
```
Comment (matiasmansilla1989) [inline app/models/target.rb]: This is already a validation that is executed on create, the name should be related to what it does and not when. Maybe, `validate_user_limit`.

**Review by matiasmansilla1989 — APPROVED** [review]

**Review by jdrosales17 — APPROVED** [review]

**Pull request merged**